packages/plugins/plugin-navtree/src/components/NavTreeContainer.tsx

```diff
@@ -75,16 +67,26 @@ export const NavTreeContainer = memo(({ tab, popoverAnchorId, topbar }: NavTreeC
   const { dispatchPromise: dispatch } = useIntentDispatcher();
   const { graph } = useAppGraph();
   const { isOpen, isCurrent, isAlternateTree, setItem } = useCapability(NavTreeCapabilities.State);
-  const layout = useLayout();
   const { navigationSidebarState } = useSidebars(NAVTREE_PLUGIN);
+  const layout = useLayout();
 
   const getActions = useCallback((node: Node) => naturalGetActions(graph, node), [graph]);
+  const getChildItems = (node?: Node, { disposition, sort = false }: TraveralOptions = {}) => {
```

Comment:
`graph.getConnections` is not reactive, this is only "working" because this function isn't wrapped in `useCallback` and so it likely forcing excessive re-renders (which is probably the source of the perf issue you mentioned)

I was also skeptical of passing a custom hook in via props, but from testing it works fine as long as its stable

Reply:
+1